cstore_fdw.c

```diff
@@ -314,36 +442,37 @@ CStoreProcessUtility(Node * parseTree, const char *queryString,
 				Assert(IsA(object, List));
 				objectName = strVal(linitial((List *) object));
 #endif
-
+				/* record cstore relation ids if cstore extension is to be dropped */
 				if (strncmp(CSTORE_FDW_NAME, objectName, NAMEDATALEN) == 0)
 				{
-					removeCStoreDirectory = true;
+					cstoreTableList = CStoreTableList();
```

Comment:
I'd open an issue to replace this with a dependency from the relation to the extension, once other parts are ready. Or at least add an XXX comment.

Reply:
I did not get this. 
There is already a dependency from cstore relations to cstore extension. CStore tables are dropped when the extension is dropped.

The issue here is keeping track of actual file nodes. If I can setup a dependency from a file node to the extension or even a relation it would be fine.  

This piece of code determines which file nodes should be dropped as a result of extension drop.

Reply:
The confusion is/was that dependencies should normally prevent that from happening. Except I guess that if you drop the extension with CASCADE, you can drop both extension and tables at the same time. But even then the tables should be dropped first, no?  Needs explanatory comment.

Reply:
added comment